### PR TITLE
Pcengines trenchboot

### PR DESCRIPTION
**Pull request by pietrushnic**

### CODE REVIEW

File: grub.cfg [hunk 1]
```diff
@@ -0,0 +1,7 @@
+set debug=linux
+serial --speed=115200 --word=8 --parity=no --stop=1
+terminal_input --append  serial
+terminal_output --append serial
+
+linux (ahci0,msdos1)/isar-image-xen-debian-buster-qemuamd64.vmlinuz-4.19.36 root=/dev/nfs rw ip=dhcp nfsroot=192.168.4.190:/srv/nfs/debian,vers=3,udp nfsrootdebug --- console=ttyS0,115200 earlyprint=serial,ttyS0,115200
```
Comment (miczyg1) [inline grub.cfg]: kernel on a hard drive with nfsroot @pietrushnic ?

Comment (pietrushnic) [reply grub.cfg]: @miczyg1 yes I placed kernel on HDD, can I somehow pull that from http server? My GRUB2-fu is limited.

Comment (miczyg1) [reply grub.cfg]: https://www.gnu.org/software/grub/manual/grub/grub.html#Network
When there is a need, there is also a solution. See also section 13.1.

Comment (pietrushnic) [reply grub.cfg]: @miczyg1 did you tested that for me or just googled? Under grub I can't see any network interfaces, do you know that issue?

File: payloads/external/Makefile.inc [hunk 2]
```diff
@@ -222,15 +222,36 @@ lzheader:
...
 lz_header-type := raw
 endif
 
+ifneq ($(CONFIG_GRUB2_INCLUDE_UROOT),)
+payloads/external/linux/linux/bzImage: linux
+paylaods/u-root/uroot/iniramfs.cpio.lzma: uroot
+
+cbfs-files-y += bzImage
+bzImage-file := payloads/external/linux/linux/bzImage
+bzImage-type := flat-binary
```
Comment (krystian-hebel) [inline payloads/external/Makefile.inc]:
```suggestion
bzImage-type := raw
```

File: payloads/external/Makefile.inc [hunk 3]
```diff
@@ -222,15 +222,36 @@ lzheader:
...
+bzImage-compression := none
+
+cbfs-files-y += initrd
+initrd-file := paylaods/u-root/uroot/iniramfs.cpio.lzma
+initrd-type := flat-binary
```
Comment (krystian-hebel) [inline payloads/external/Makefile.inc]:
```suggestion
initrd-type := raw
```

**Comment by pietrushnic** [comment]
@miczyg1 @krystian-hebel seem to work https://pastebin.com/siwYM80E

**Comment by pietrushnic** [comment]
@miczyg1 @krystian-hebel do you think we have here anything to contribute to upstream projects?

**Comment by miczyg1** [comment]
@pietrushnic we can upstream the landing zone to the coreboot's security directory for example? Intel TXT support is landing there. We could take the maintainership responsibility for the LZ in coreboot.

In terms of u-root and Linux, there is LinuxBoot already which seems to do a very similar thing. bad thing is, that I realized about it quite late.

**Comment by pietrushnic** [comment]
@miczyg1 ideally would be if we can consult that with @dpsmith

**Comment by miczyg1** [comment]
@pietrushnic here's a newer branch with SHA256 support: https://github.com/pcengines/coreboot/tree/pcengines_trenchboot_v4.10.0.2

**Comment by pietrushnic** [comment]
@miczyg1 I'm closing this just to get rid of possible further confusion

**Pull request closed without merging**